plugins/plugin-chart-echarts/src/Pie/index.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { t } from '@superset-ui/translation';
+import { ChartMetadata, ChartPlugin } from '@superset-ui/chart';
+import buildQuery from './buildQuery';
+import controlPanel from './controlPanel';
+import transformProps from './transformProps';
+import thumbnail from './images/thumbnail.png';
+
+const metadata = new ChartMetadata({
+  description: 'Pie chart using ECharts',
+  name: t('EchartsPie'),
```

Comment:
by creating the metadata here instead of inside the constructor of the plugin, you need to ensure the translator is initialized **before importing this file**. 

Better create the metadata inside the constructor of the plugin

Reply:
It appears this may be a problem in several plugins, as most (all?) seem to define metadata outside the constructor. I'll leave this as-if for now and clean up all plugins in one sweep (we also need to fix the yeoman generator).

Reply:
I have added the fix in the newer plugins. 
https://github.com/apache-superset/superset-ui/blob/master/plugins/preset-chart-xy/src/BoxPlot/index.ts